todos/src/reducers/todoReducers.js

```diff
@@ -46,7 +47,54 @@ const todoReducers = (state = initialData, action) => {
         ...state,
         list: updatedList,
       };
+    case "EDIT_TODO":
+      const updatedList2 = state.list.map((todo) => {
+        if (todo.id === action.id) {
+          return {
+            ...todo,
+            onEdit: action.onEdit,
+          };
+        }
+        return todo;
+      });
+
+      return {
+        ...state,
+        list: updatedList2,
+      };
+    case "EDIT_UPDATE_COMPLETED":
+      const editUdupdatedList = state.list.map((todo) => {
+        if (todo.id === action.id) {
+          return {
+            ...todo,
+            isCompleted: action.isCompleted,
+            date: action.date,
+            completedDate: action.completedDate,
+            onEdit: !action.onEdit,
+          };
+        }
+        return todo;
+      });
 
+      return {
+        ...state,
+        list: editUdupdatedList,
+      };
+    case "UPDATE_TODO":
+      const updatedList3 = state.list.map((todo) => {
```

Comment:
Can we use the same name for each case as they have individual scopes?

Reply:
@hhu-wtag  Bhai, I tried to make case names more meaningful by renaming some of them.